File: day20/人以群分/人以分群.go
package main
//我丢这个都要超时,没办法go语言一般的读写太慢了
import (
	"fmt"
	"sort"
)

func main() {
	var i, n, sum int
	fmt.Scanln(&n)
	var sli []int = make([]int, n)
	for i = 0; i < n; i++ {
		fmt.Scanf("%d", &sli[i])
	}
	sort.Ints(sli)
	for i = 0; i < n/2; i++ {
		sum -= sli[i]
	}
	for i < n {
		sum += sli[i]
		i++
	}
	fmt.Println("Outgoing #:", n/2+n%2)
	fmt.Println("Introverted #:", n/2)
	fmt.Println("Diff =", sum)
}

File: day20/人以群分/人以分群.cpp
#include<cstdio>
#include<algorithm>
#define MAXSIZE 100000
using namespace std;
int arr[MAXSIZE];
int main() {
    int n, i, sum = 0;
    scanf("%d",&n);
    for(i = 0; i < n; i++) {
        scanf("%d",&arr[i]);
    }
    sort(arr,arr + n);
    for(i = 0; i < n / 2; i++) {
        sum -= arr[i];
    }
    while(i < n) {
        sum += arr[i++];
    }
    printf("Outgoing #: %d\n", n/2+n%2);
	printf("Introverted #: %d\n", n/2);
	printf("Diff = %d\n", sum);
    return 0;
}